backend/controller/door.go

```diff
@@ -17,6 +17,25 @@ func GetAllDoors(w http.ResponseWriter, _ *http.Request) {
 	helpers.JsonWriter(w, doors)
 }
 
+func UpdateDoor(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	err, dId := helpers.ParseInt(w, r, "doorId")
+	if err != nil {
+		return
+	}
+	err, door := helpers.GetFirstTable(w, models.Door{}, models.Common{Id: dId})
+	if err != nil {
+		return
+	}
+	err, door = helpers.UpdateSpecifiedParams(w, r, &door, door.Common, &door.Common)
```

Comment:
the pointer shit here is like kinda confusing, but essentially im just using it as an agnostic way to update the common variable in the helper function even though we can't explicity call common, to check if the user passed any values to common

Reply:
I already looked into readonly fields, it doesn't work for id or CreatedAt as far as I can tell, because CreatedAt will not be updated on post if it is read only and making id readonly breaks.

Couple of other things, a lot of this is just like limitations with Golang itself, like in most languages you can support generic interfaces that all have a common subset and its just weird in go. 

Also even if I removed common entirely here,  the only thing that would change is you wouldn't have a pointer. Esssentially the pointer is saying whenever you update door (via copying the input values to the found door object), also update "updatedCommon" (see helper function) so that the static "initCommon" can be compared to see if there were any common values. The main issue is the interface we have "T" will not let us do T.Common, even though every object within T contains common, there's no real golang workaround for that. However I will say, this pointer method once u like take 2 seconds thinking abt pointers does make sense and I can reuse this to handle all future issues with Common comparisons as this has been my major issue how to generically pass a common function that is changing.

I haven't looked into embedded, didn't see anything written on it when I was looking through solutions to our problems so I also don't think it'll help because we are trying to solve a typing issue not a GORM issue, but I can see.

I could in theory bring back the "reqUser" and "reqDoors" I did see a json marshal, unmarshal method that may be promising to go from reqUser->User this way reqUser will only contain the editable fields, but it does add back in a decent amount of extra boilerplate hence why I didn't immediately do it when I had this working.

Reply:
I find it hard to believe gorm doesn't support a better way to do this, especially because our relations are not particularly complicated.

At least we should abstract the common stuff away from whoever is using these functions as much as possible, I don't see why we can't implement this logic in the function body instead of ask for the user to pass in more parameters.

Reply:
I agree that updating Common is making this a bit complicated when it was initially supposed to do the opposite.

Do we actually need `CreatedAt` or `UpdatedAt` anywhere? If there's no value in keeping them then I think we should get rid of Common to make all of this more readable.

Reply:
not sure about usage, I do know that comparing id is going to be the exact same as comparing common (generically) cuz essentially this whole comparing commons still non generically would boil down to 1 if statement. Unless you know of some way i don't to compare it

Reply:
Ah, you're right.
Maybe it's still worth doing if we're not using `CreatedAt` or `UpdatedAt`. Then we won't need to worry about if they should/shouldn't be read only.
I guess adding a method to get `id` in the interface might be worthwhile if you're doing it a lot.

Reply:
if that works lol, tried doing something similar with common